sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -253,6 +259,12 @@ class JacksonParser(
           parser.getLongValue * 1000000L
       }
 
+    case TimestampNTZType =>
+      (parser: JsonParser) => parseJsonToken[java.lang.Long](parser, dataType) {
+        case VALUE_STRING if parser.getTextLength >= 1 =>
+          timestampNTZFormatter.parseWithoutTimeZone(parser.getText)
```

Comment:
Casting from numerical types to `timestamp_ntz` seems not supported so I don't care `VALUE_NUMBER_INT` here.